### PR TITLE
Add flag for replication cleanup process

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
Add a feature flag to config rpc replication queue clean up process

**Why?**
Disable this cleanup so that we don't have backward compatibility if we want to rollback

**How did you test it?**
Local test

**Potential risks**
The task deletes by the process and after rolled back, the original logic cannot find the task and throws internal service error

### CODE REVIEW

File: service/frontend/service.go [hunk 1]
```diff
@@ -135,6 +136,7 @@ func NewConfig(dc *dynamicconfig.Collection, numHistoryShards int, enableReadFro
 		DisallowQuery:                       dc.GetBoolPropertyFnWithDomainFilter(dynamicconfig.DisallowQuery, false),
 		SendRawWorkflowHistory:              dc.GetBoolPropertyFnWithDomainFilter(dynamicconfig.SendRawWorkflowHistory, false),
 		EnableRPCReplication:                dc.GetBoolProperty(dynamicconfig.FrontendEnableRPCReplication, false),
+		EnableCleanupReplicationTask:        dc.GetBoolProperty(dynamicconfig.FrontendEnableCleanupReplicationTask, false),
```
Comment (yycptt) [inline service/frontend/service.go:139]: Do we need to default to `true` as currently the code will always do clean up? Same for the flag for history service.

File: service/frontend/adminHandler.go [hunk 2]
```diff
@@ -117,12 +117,16 @@ func (adh *AdminHandler) RegisterHandler() {
...
+		adh.Resource.GetDomainReplicationQueue().Start()
+	}
 }
 
 // Stop stops the handler
 func (adh *AdminHandler) Stop() {
-	adh.Resource.GetDomainReplicationQueue().Stop()
+	if adh.config.EnableCleanupReplicationTask() {
+		adh.Resource.GetDomainReplicationQueue().Stop()
+	}
```
Comment (yycptt) [inline service/frontend/adminHandler.go:129]: The value for this dynamicconfig may and will change when the service is running, so it's possible only `Start()` is called or only `Stop()` is called. Ideally the config value should be get once and stored somewhere on service startup.

I checked the implementation for domain replication queue, it seems to be ok if only `Stop()` is called, not sure about what will happen if only `Start()` is called. But if you're relying on this fact, probably add some comments in the code.

BTW, there's a copy & paste error on `domainReplicationQueue.go` L104.

Comment (yux0) [reply service/frontend/adminHandler.go:129]: Good point. So I updated it to always call stop() and check the flag before calling start()

File: service/frontend/service.go (hunk 1 above)
Comment (yux0) [reply service/frontend/service.go:139]: update fault to true

File: common/persistence/domainReplicationQueue.go [hunk 3]
```diff
@@ -101,7 +101,7 @@ func (q *domainReplicationQueueImpl) Start() {
 }
 
 func (q *domainReplicationQueueImpl) Stop() {
-	if !atomic.CompareAndSwapInt32(&q.status, common.DaemonStatusInitialized, common.DaemonStatusStarted) {
+	if !atomic.CompareAndSwapInt32(&q.status, common.DaemonStatusInitialized, common.DaemonStatusStopped) {
```
Comment (yycptt) [inline common/persistence/domainReplicationQueue.go:104]: `common.DaemonStatusInitialized` -> `common.DaemonStarted`

File: service/frontend/adminHandler.go (hunk 2 above)
Comment (yycptt) [reply service/frontend/adminHandler.go]: Please also add some comments explaining that it's ok to call `Stop()` without calling `Start()`.

**Pull request merged**